src/java/org/apache/cassandra/index/sai/SSTableIndex.java

```diff
@@ -73,13 +80,35 @@ public SSTableIndex(SSTableContext sstableContext, IndexComponents.ForRead perIn
     {
         assert perIndexComponents.context().getValidator() != null;
         this.perIndexComponents = perIndexComponents;
-        this.searchableIndex = perIndexComponents.version().onDiskFormat().newSearchableIndex(sstableContext, perIndexComponents);
+        this.searchableIndex = createSearchableIndex(sstableContext, perIndexComponents);
 
         this.sstableContext = sstableContext.sharedCopy(); // this line must not be before any code that may throw
         this.indexContext = perIndexComponents.context();
         this.sstable = sstableContext.sstable;
     }
 
+    private static SearchableIndex createSearchableIndex(SSTableContext sstableContext, IndexComponents.ForRead perIndexComponents)
+    {
+        if (CassandraRelevantProperties.SAI_INDEX_READS_DISABLED.getBoolean())
+        {
+            logger.info("Creating dummy index searcher for sstable {} as SAI index reads are disabled", sstableContext.sstable.descriptor);
```

Comment:
this could be spammy on compactor nodes
I would log it at TRACE level

Reply:
This happens once for every opened sstable, and we already log a bunch of per-sstable message at info/debug, so I doubt this is a problem. And this feel kind of important to me to see that this happen, and particularly initially to know that this is working as we deploy it (and switching to trace is a PITA).

Anyway, that was useful to me to check this worked as I integrated it in CNDB and I think that's fine/useful, but I don't care any more than that and happy to remove (or move to trace, but to me trace is useless and trace logs are just pollution).

Reply:
+1 This per sstable is fine and useful